<launch>
	<param name="/use_sim_time" value="true"/>

	<!-- OPENNI NODE: Our bag files record depth and rgb images, this is required to produce point clouds -->
   <include file="$(find openni_launch)/launch/openni.launch">
   	    <arg name="load_driver" value="false" />
   </include>

  <node name="omnimapper_ros_node" pkg="omnimapper_ros" type="omnimapper_ros_node" respawn="false" output="screen">
  	
    <!-- PLUGIN SELECTION -->
    <param name="use_icp" value="true"/>
    <param name="use_csm" value="true"/>
    <param name="use_occ_edge_icp" value="false"/>
    <param name="use_tf" value="true"/>
    <param name="use_no_motion" value="false"/>
    <param name="use_error_plugin" value="false"/>
    <param name="use_tsdf_plugin" value="false"/>
    <param name="use_organized_segmentation" value="false"/>

    <!-- INITIALIZATION -->
    <param name="use_init_pose" value="false"/>	
    <param name="init_pose_from_tf" value="true"/>
  	<param name="odom_frame_name" value="/odom"/>
    <param name="base_frame_name" value="/base_link"/>
    <param name="cloud_topic_name" value="/camera/depth_registered/points"/>

    <!-- ICP PLUGIN -->
    <param name="icp_leaf_size" value="0.05"/>
    <param name="icp_max_correspondence_distance" value="2.2"/>
    <param name="icp_trans_noise" value="0.15"/>
    <param name="icp_rot_noise" value="0.15"/>
    <param name="icp_loop_closure_distance_threshold" value="3.0"/>
    <param name="icp_score_threshold" value="0.8"/>
    <param name="icp_loop_closure_score_threshold" value="0.8"/>
    <param name="icp_loop_closure_pose_index_threshold" value="10"/>
    <param name="icp_save_full_res_clouds" value="false"/>

    <!-- TF PLUGIN -->
    <param name="tf_trans_noise" value="0.05"/>
    <param name="tf_roll_noise" value="0.001"/>
    <param name="tf_pitch_noise" value="0.001"/>
    <param name="tf_yaw_noise" value="0.2"/>

    <!-- VISUALIZATION -->
    <param name="broadcast_map_to_odom" value="true"/>
    <param name="draw_pose_array" value="true"/>
    <param name="draw_pose_graph" value="true"/>
    <param name="draw_label_cloud" value="true"/>
    <param name="draw_clusters" value="true"/>

    <!-- CONFIG -->
   <param name="suppress_commit_window" value="false"/>
   <param name="add_pose_per_cloud" value="false"/>
  </node>

</launch>